File: app/src/main/java/id/co/blogbasbas/wisatasemarang/CreateActivity.java
package id.co.blogbasbas.wisatasemarang;

import android.Manifest;
import android.app.ProgressDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.provider.MediaStore;
import android.support.annotation.NonNull;
import android.support.design.widget.TextInputLayout;
import android.support.v7.app.AlertDialog;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.android.gms.common.GooglePlayServicesNotAvailableException;
import com.google.android.gms.common.GooglePlayServicesRepairableException;
import com.google.android.gms.location.places.Place;
import com.google.android.gms.location.places.ui.PlacePicker;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;
import java.io.IOException;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import id.co.blogbasbas.wisatasemarang.kumpulanhelper.MyFunction;
import id.co.blogbasbas.wisatasemarang.kumpulanhelper.RealPathUtils;
import id.co.blogbasbas.wisatasemarang.network.ApiServices;
import id.co.blogbasbas.wisatasemarang.network.RetrofitConfig;
import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import pub.devrel.easypermissions.EasyPermissions;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class CreateActivity extends MyFunction implements EasyPermissions.PermissionCallbacks {

    @BindView(R.id.imageView)
    ImageView imageView;
    @BindView(R.id.edt_deskripsi)
    EditText edtDeskripsi;
    @BindView(R.id.input_layout_deskripsi)
    TextInputLayout inputLayoutDeskripsi;
    @BindView(R.id.edt_nama)
    EditText edtNama;
    @BindView(R.id.input_layout_nama)
    TextInputLayout inputLayoutNama;
    @BindView(R.id.edt_alamat)
    EditText edtAlamat;
    @BindView(R.id.input_layout_alamat)
    TextInputLayout inputLayoutAlamat;
    @BindView(R.id.edt_event)
    EditText edtEvent;
    @BindView(R.id.input_layout_event)
    TextInputLayout inputLayoutEvent;
    @BindView(R.id.btn_maps)
    Button btnMaps;
    @BindView(R.id.status_maps)
    TextView statusMaps;
    @BindView(R.id.btn_submit)
    Button btnSubmit;

    ProgressDialog loading;
    private static final String TAG = "CreateActivity";
    private static final int RC_GALLERY = 3;
    private static final int RC_MAPS = 4;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_create);
        ButterKnife.bind(this);
    }

    @OnClick({R.id.imageView, R.id.btn_maps, R.id.btn_submit})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.imageView:
                //dapet foto dari hp
                openGallery();
                break;
            case R.id.btn_maps:
                PlacePicker.IntentBuilder builder = new PlacePicker.IntentBuilder();
                try {
                    startActivityForResult(builder.build(CreateActivity.this), RC_MAPS);
                } catch (GooglePlayServicesNotAvailableException e) {
                    e.printStackTrace();
                } catch (GooglePlayServicesRepairableException e) {
                    e.printStackTrace();
                }
                break;
            case R.id.btn_submit:
                //ambil data
                String nama, deskripsi, alamat, event;
                nama = edtNama.getText().toString();
                alamat = edtAlamat.getText().toString();
                deskripsi = edtDeskripsi.getText().toString();
                event = edtEvent.getText().toString();

                //cek jika kosong
                if (TextUtils.isEmpty(nama) ||
                        TextUtils.isEmpty(alamat)||
                        TextUtils.isEmpty(deskripsi)||
                        TextUtils.isEmpty(event)||
                        TextUtils.isEmpty(lat)||
                        TextUtils.isEmpty(lng)||
                        uri == null
                        ){
                    MyToast("belum diisi");
                } else {

                    //dapetin path gambar yg dipilih
                    String path = null;
                    if (Build.VERSION.SDK_INT < 11) {
                        path = RealPathUtils.getRealPathFromURI_BelowAPI11(CreateActivity.this, uri);

                        // SDK >= 11 && SDK < 19
                    } else if (Build.VERSION.SDK_INT < 19) {
                        path = RealPathUtils.getRealPathFromURI_API11to18(CreateActivity.this, uri);

                        // SDK > 19 (Android 4.4)
                    } else if (Build.VERSION.SDK_INT > 22) {
                        path = RealPathUtils.getRealPathFromURI_API23(CreateActivity.this, uri);

                    } else {
                        path = RealPathUtils.getRealPathFromURI_API19(CreateActivity.this, uri);
                    }

                    File file = new File(path);

                    //ngasih nama gambar
                    String namaGambar = currentDate() + file.getName();

                    RequestBody sFile = RequestBody.create(MediaType.parse("image/*"), file);
                    MultipartBody.Part fileToUpload = MultipartBody.Part.createFormData("file", namaGambar, sFile);
                    RequestBody sNama = RequestBody.create(MediaType.parse("text/plain"), nama);
                    RequestBody sdeskripsi = RequestBody.create(MediaType.parse("text/plain"), deskripsi);
                    RequestBody sevent = RequestBody.create(MediaType.parse("text/plain"), event);
                    RequestBody salamat = RequestBody.create(MediaType.parse("text/plain"), alamat);
                    RequestBody sgambar = RequestBody.create(MediaType.parse("text/plain"), namaGambar);
                    RequestBody slat = RequestBody.create(MediaType.parse("text/plain"), lng);
                    RequestBody slng = RequestBody.create(MediaType.parse("text/plain"), lat);

                    //tampilkan loading
                    loading = ProgressDialog.show(CreateActivity.this, "Loading", "Mohon bersabara", true, false);
                    loading.show();

                    //persiapan service
                    ApiServices api = RetrofitConfig.getApiServices();
                    api.CREATE_WISATA(
                            fileToUpload,
                            sNama,
                            sgambar,
                            sdeskripsi,
                            sevent,
                            slat,
                            slng,
                            salamat
                    ).enqueue(new Callback<ResponseBody>() {
                        @Override
                        public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                            if (response.isSuccessful()) {
                                loading.dismiss();
                                try {
                                    JSONObject object = new JSONObject(response.body().string());
                                    if (object.getString("success").equals("true")) {

                                        Log.d(TAG, "onResponse: ");
                                        AlertDialog.Builder builder = new AlertDialog.Builder(CreateActivity.this);
                                        builder.setTitle("Complete");
                                        builder.setMessage(object.getString("message"));
                                        builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                                            public void onClick(DialogInterface dialog, int id) {
                                                dialog.dismiss();
                                                Intent a = new Intent(CreateActivity.this, MainActivity.class);
                                                a.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                                                startActivity(a);
                                                finish();
                                            }
                                        });
                                        builder.create().show();

                                    } else {
                                        MyToast("Gagal ");
                                    }
                                } catch (JSONException e) {
                                    e.printStackTrace();
                                    MyToast("gagal 1 " +e);
                                    Log.d(TAG, "gagal upload : " +e);
                                } catch (IOException e) {
                                    MyToast("gagal 2 " +e);
                                    e.printStackTrace();
                                }
                            } else {
                                MyToast("Not Succesfull" +response);
                            }
                        }


                        @Override
                        public void onFailure(Call<ResponseBody> call, Throwable t) {
                            MyToast("Gagal Failure" +t );
                        }
                    });
                }

                    break;

        }
    }

    private void openGallery() {

        String[] perms = {Manifest.permission.READ_EXTERNAL_STORAGE};
        if (!EasyPermissions.hasPermissions(getApplicationContext(), perms)) {
            EasyPermissions.requestPermissions(CreateActivity.this, "Butuh kamu", 100, perms);
        }

        Intent gallery = new Intent();
        gallery.setType("image/*");
        gallery.setAction(Intent.ACTION_GET_CONTENT);
        startActivityForResult(gallery, RC_GALLERY);
    }

    public Uri uri = null;
    String lat, lng;
    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == RC_GALLERY && resultCode == RESULT_OK && data != null) {
            uri = data.getData();
            Log.d(TAG, "onActivityResult: " + uri.toString());
            try {
                Bitmap hasil = MediaStore.Images.Media.getBitmap(getContentResolver(), uri);
                imageView.setImageBitmap(hasil);
            } catch (IOException e) {
                e.printStackTrace();
            }
        } else if (requestCode == RC_MAPS && resultCode  == RESULT_OK){
            Place place = PlacePicker.getPlace(CreateActivity.this, data);
            String alamat = String.format("%s", place.getName());
            lat = String.valueOf(place.getLatLng().latitude);
            lng = String.valueOf(place.getLatLng().longitude);
            statusMaps.setText(alamat);
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);

        // Forward results to EasyPermissions
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
    }

    @Override
    public void onPermissionsGranted(int requestCode, List<String> perms) {
        if (requestCode == 100) {
            openGallery();
        }
    }

    @Override
    public void onPermissionsDenied(int requestCode, List<String> perms) {

    }
}

File: app/src/main/java/id/co/blogbasbas/wisatasemarang/CreateActivity.kt
package id.co.blogbasbas.wisatasemarang

import android.Manifest
import android.annotation.TargetApi
import android.app.Activity
import android.app.ProgressDialog
import android.content.ContentUris
import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Environment
import android.provider.DocumentsContract
import android.provider.MediaStore
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import com.bumptech.glide.Glide
import com.google.android.gms.common.GooglePlayServicesNotAvailableException
import com.google.android.gms.common.GooglePlayServicesRepairableException
import com.google.android.gms.location.places.ui.PlacePicker
import com.google.android.material.textfield.TextInputLayout
import com.karumi.dexter.Dexter
import com.karumi.dexter.MultiplePermissionsReport
import com.karumi.dexter.PermissionToken
import com.karumi.dexter.listener.PermissionRequest
import com.karumi.dexter.listener.multi.MultiplePermissionsListener
import id.co.blogbasbas.wisatasemarang.kumpulanhelper.MyFunction
import id.co.blogbasbas.wisatasemarang.network.RetrofitConfig
import kotlinx.android.synthetic.main.activity_create.*
import okhttp3.MediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody
import org.json.JSONException
import pl.aprilapps.easyphotopicker.EasyImage
import pub.devrel.easypermissions.EasyPermissions
import pub.devrel.easypermissions.EasyPermissions.PermissionCallbacks
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.io.File
import java.io.IOException

class CreateActivity : MyFunction(), PermissionCallbacks, View.OnClickListener {
    var imageView: ImageView? = null
    var edtDeskripsi: EditText? = null
    var inputLayoutDeskripsi: TextInputLayout? = null
    var edtNama: EditText? = null
    var inputLayoutNama: TextInputLayout? = null
    var edtAlamat: EditText? = null
    var inputLayoutAlamat: TextInputLayout? = null
    var edtEvent: EditText? = null
    var inputLayoutEvent: TextInputLayout? = null
    var btnMaps: Button? = null
    var statusMaps: TextView? = null
    var btnSubmit: Button? = null
    var loading: ProgressDialog? = null
    var easyImage: EasyImage? = null
    var fileFoto = 0


    var path: File? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_create)
        imageView = findViewById(R.id.imageView)
        edtDeskripsi = findViewById(R.id.edt_deskripsi)
        edtAlamat = findViewById(R.id.edt_alamat)
        edtNama = findViewById(R.id.edt_nama)
        edtEvent = findViewById(R.id.edt_event)
        statusMaps = findViewById(R.id.status_maps)
        btnMaps = findViewById(R.id.btn_maps)
        btnSubmit = findViewById(R.id.btn_submit)

        Dexter.withActivity(this).withPermissions(
                Manifest.permission.READ_EXTERNAL_STORAGE,
                Manifest.permission.WRITE_EXTERNAL_STORAGE
        )
                .withListener(object : MultiplePermissionsListener {
                    override fun onPermissionsChecked(report: MultiplePermissionsReport?) {
                        report?.let {

                        }
                    }

                    override fun onPermissionRationaleShouldBeShown(
                            permissions: MutableList<PermissionRequest>?,
                            token: PermissionToken?
                    ) {
                        token?.continuePermissionRequest()
                    }
                }).check()
        btn_submit.setOnClickListener {
            submitData()
        }
        btn_maps.setOnClickListener {
            val builder = PlacePicker.IntentBuilder()
            try {
                startActivityForResult(builder.build(this@CreateActivity), RC_MAPS)
            } catch (e: GooglePlayServicesNotAvailableException) {
                e.printStackTrace()
            } catch (e: GooglePlayServicesRepairableException) {
                e.printStackTrace()
            }
        }

        edt_empat?.setOnClickListener {
           initPermission()
            openGallery()
            fileFoto = 4


        }

        edt_satu.setOnClickListener {
            initPermission()
            openGallery()
            fileFoto = 1
        }
        edt_dua.setOnClickListener {
            openGallery()
            fileFoto = 2
        }
        edt_tiga.setOnClickListener {
            openGallery()
            fileFoto = 3
        }

    }

    fun uploadFoto() {
        val file = File(path?.toURI())
        //ngasih nama gambar
        val namaGambar = file.name
        val sFile = RequestBody.create(MediaType.parse("image/*"), file)
        val fileToUpload = MultipartBody.Part.createFormData("file", namaGambar, sFile)
        val sgambar = RequestBody.create(MediaType.parse("text/plain"), namaGambar)

        //tampilkan loading

        progress_circular.visibility = View.VISIBLE
        //persiapan service
        val api = RetrofitConfig.apiServices
        api.CREATE_FOTO(
                fileToUpload,
                sgambar

        ).enqueue(object : Callback<id.co.blogbasbas.wisatasemarang.model.ResponseBody> {
            override fun onResponse(call: Call<id.co.blogbasbas.wisatasemarang.model.ResponseBody>, response: Response<id.co.blogbasbas.wisatasemarang.model.ResponseBody>) {
                if (response.isSuccessful) {
                    progress_circular.visibility = View.GONE

                    try {

                        if (response.body()?.success?.equals(true)!!) {
                            Log.e(TAG, "onResponse: ")
                            val builder = AlertDialog.Builder(this@CreateActivity)
                            builder.setTitle("Sukses Upload gambar")
                            builder.setMessage("Gambar $fileFoto Sukses  di upload")
                            builder.setPositiveButton("OK") { dialog, id ->
                                dialog.dismiss()

                            }
                            builder.create().show()
                        } else {
                            MyToast("gambar sudah tersedia")
                        }
                    } catch (e: JSONException) {
                        e.printStackTrace()
                        MyToast("gagal 1 $e")
                        Log.e(TAG, "gagal upload : $e")
                    } catch (e: IOException) {
                        MyToast("gagal 2 $e")
                        e.printStackTrace()
                    }
                } else {
                    MyToast("Not Succesfull$response")
                }
            }

            override fun onFailure(call: Call<id.co.blogbasbas.wisatasemarang.model.ResponseBody>, t: Throwable) {
                MyToast("Gagal Failure$t")
                Log.e("TAG", "error ${t.message}")
                progress_circular.visibility = View.GONE

            }
        })
    }


    private fun initPermission() {

        Dexter.withActivity(this).withPermissions(
                Manifest.permission.CAMERA,
                Manifest.permission.READ_EXTERNAL_STORAGE,
                Manifest.permission.WRITE_EXTERNAL_STORAGE
        )
                .withListener(object : MultiplePermissionsListener {
                    override fun onPermissionsChecked(report: MultiplePermissionsReport?) {
                        report?.let {

                        }
                    }

                    override fun onPermissionRationaleShouldBeShown(
                            permissions: MutableList<PermissionRequest>?,
                            token: PermissionToken?
                    ) {
                        token?.continuePermissionRequest()
                    }
                }).check()
    }

    fun submitData() {
        //ambil data
        val nama: String
        val deskripsi: String
        val alamat: String
        val event: String

        nama = edtNama!!.text.toString()
        alamat = edtAlamat!!.text.toString()
        deskripsi = edtDeskripsi!!.text.toString()
        event = edtEvent!!.text.toString()
        val namaGambar1 = edt_satu.text.toString()
        val namaGambar2 = edt_dua.text.toString()
        val namaGambar3 = edt_tiga.text.toString()
        val namaGambar4 = edt_empat.text.toString()
        //cek jika kosong
        if (
                TextUtils.isEmpty(nama) ||
                TextUtils.isEmpty(alamat) ||
                TextUtils.isEmpty(deskripsi) ||
                TextUtils.isEmpty(event) ||
                TextUtils.isEmpty(lat) ||
                TextUtils.isEmpty(lng) ||
                TextUtils.isEmpty(namaGambar1) ||
                TextUtils.isEmpty(namaGambar2) ||
                TextUtils.isEmpty(namaGambar3) ||
                TextUtils.isEmpty(namaGambar4)
        //uri == null

        ) {
            MyToast("belum diisi")
        } else { //dapetin path gambar yg dipilih

            val file = File(path?.toURI())
            //ngasih nama gambar
            val namaGambar = file.name

            val sFile = RequestBody.create(MediaType.parse("image/*"), file)
            val fileToUpload = MultipartBody.Part.createFormData("file", namaGambar, sFile)
            val sNama = RequestBody.create(MediaType.parse("text/plain"), nama)
            val sdeskripsi = RequestBody.create(MediaType.parse("text/plain"), deskripsi)
            val sevent = RequestBody.create(MediaType.parse("text/plain"), event)
            val salamat = RequestBody.create(MediaType.parse("text/plain"), alamat)
            val sgambar1 = RequestBody.create(MediaType.parse("text/plain"), namaGambar1)
            val sgambar2 = RequestBody.create(MediaType.parse("text/plain"), namaGambar2)
            val sgambar3 = RequestBody.create(MediaType.parse("text/plain"), namaGambar3)
            val sgambar4 = RequestBody.create(MediaType.parse("text/plain"), namaGambar4)
            val slat = RequestBody.create(MediaType.parse("text/plain"), lng)
            val slng = RequestBody.create(MediaType.parse("text/plain"), lat)
            //tampilkan loading
            progress_circular.visibility = View.VISIBLE
            //persiapan service
            val api = RetrofitConfig.apiServices
            api.CREATE_WISATA(
                    //fileToUpload,
                    sNama,
                    sgambar1,
                    sdeskripsi,
                    sevent,
                    slat,
                    slng,
                    salamat,
                    sgambar2,
                    sgambar3,
                    sgambar4
            ).enqueue(object : Callback<id.co.blogbasbas.wisatasemarang.model.ResponseBody> {
                override fun onResponse(call: Call<id.co.blogbasbas.wisatasemarang.model.ResponseBody>, response: Response<id.co.blogbasbas.wisatasemarang.model.ResponseBody>) {
                    if (response.isSuccessful) {
                        progress_circular.visibility = View.GONE

                        try {

                            if (response.body()?.success?.equals(true)!!) {
                                Log.d(TAG, "onResponse: ")
                                val builder = AlertDialog.Builder(this@CreateActivity)
                                builder.setTitle("Complete")
                                builder.setMessage(response.body()?.message)
                                builder.setPositiveButton("OK") { dialog, id ->
                                    dialog.dismiss()
                                    val a = Intent(this@CreateActivity, MainActivity::class.java)
                                    a.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
                                    startActivity(a)
                                    finish()
                                }
                                builder.create().show()
                            } else {
                                MyToast("Gagal ${response.body()?.message}")
                            }
                        } catch (e: JSONException) {
                            e.printStackTrace()
                            MyToast("gagal 1 $e")
                            Log.d(TAG, "gagal upload : $e")
                        } catch (e: IOException) {
                            MyToast("gagal 2 $e")
                            e.printStackTrace()
                        }
                    } else {
                        MyToast("Not Succesfull$response")
                    }
                }

                override fun onFailure(call: Call<id.co.blogbasbas.wisatasemarang.model.ResponseBody>, t: Throwable) {
                    MyToast("Gagal Failure$t")
                    Log.e("TAG", "error ${t.message}")
                    progress_circular.visibility = View.GONE

                }
            })
        }
    }

    private val OPERATION_CHOOSE_PHOTO = 2
    private fun openGallery() {


        val intent = Intent(Intent.ACTION_GET_CONTENT)
        intent.type = "image/*"
        startActivityForResult(intent, OPERATION_CHOOSE_PHOTO)


    }

    var uri: Uri? = null
    var lat: String? = null
    var lng: String? = null

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == OPERATION_CHOOSE_PHOTO && resultCode == Activity.RESULT_OK && data != null) {
            if (Build.VERSION.SDK_INT >= 19) {
                handleImageOnKitkat(data)
                uri = data.getData();
            }
        } else if (requestCode == RC_MAPS && resultCode == Activity.RESULT_OK) {
            val place = PlacePicker.getPlace(this@CreateActivity, data)
            val alamat = String.format("%s", place.name)
            lat = place.latLng.latitude.toString()
            lng = place.latLng.longitude.toString()
            statusMaps!!.text = alamat
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        // Forward results to EasyPermissions
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this)
    }

    override fun onPermissionsGranted(requestCode: Int, perms: List<String>) {
        if (requestCode == 100) {
            openGallery()
        }
    }

    private fun getImagePath(uri: Uri?, selection: String?): String {
        var path: String? = null
        val cursor = uri?.let { contentResolver.query(it, null, selection, null, null) }
        if (cursor != null) {
            if (cursor.moveToFirst()) {
                path = cursor.getString(cursor.getColumnIndex(MediaStore.Images.Media.DATA))
            }
            cursor.close()
        }
        return path!!
    }

    @TargetApi(19)
    private fun handleImageOnKitkat(data: Intent?) {
        var imagePath: String? = null
        val uri = data!!.data
        //DocumentsContract defines the contract between a documents provider and the platform.
        if (DocumentsContract.isDocumentUri(this, uri)) {
            val docId = DocumentsContract.getDocumentId(uri)
            if (uri != null) {
                if ("com.android.providers.media.documents" == uri.authority) {
                    val id = docId.split(":")[1]
                    val selsetion = MediaStore.Images.Media._ID + "=" + id
                    imagePath = getImagePath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
                            selsetion)
                } else if ("com.android.providers.downloads.documents" == uri.authority) {
                    val contentUri = ContentUris.withAppendedId(Uri.parse(
                            "content://downloads/public_downloads"), java.lang.Long.valueOf(docId))
                    imagePath = getImagePath(contentUri, null)
                }
            }
        } else if (uri != null) {
            if ("content".equals(uri.scheme, ignoreCase = true)) {
                imagePath = getImagePath(uri, null)
            } else if (uri != null) {
                if ("file".equals(uri.scheme, ignoreCase = true)) {
                    imagePath = uri.path
                }
            }
        }

        var file = File(imagePath.toString())
        val compressedImage = id.zelory.compressor.Compressor(this)
                .setMaxWidth(640)
                .setMaxHeight(480)
                .setQuality(75)
                .setCompressFormat(Bitmap.CompressFormat.PNG)
                .setDestinationDirectoryPath(
                        Environment.getExternalStoragePublicDirectory(
                                Environment.DIRECTORY_PICTURES
                        ).absolutePath
                )
                .compressToFile(file)
        path = compressedImage
        imageView?.let { Glide.with(this).load(compressedImage).into(it) }
        val files = File(path?.toURI())
        val namaGambar = files.name

        when (fileFoto) {
            1 -> {
                edt_satu.setText(namaGambar)
                uploadFoto()
            }
            2 -> {
                edt_dua.setText(namaGambar)
                uploadFoto()
            }
            3 -> {
                edt_tiga.setText(namaGambar)
                uploadFoto()
            }
            4 -> {
                edt_empat.setText(namaGambar)
                uploadFoto()
            }

        }

    }

    private fun chooseFile(file: File) {

    }

    override fun onPermissionsDenied(requestCode: Int, perms: List<String>) {}
    override fun onClick(view: View) {

    }

    companion object {
        private const val TAG = "CreateActivity"
        private const val RC_GALLERY = 3
        private const val RC_MAPS = 4
    }
}